compiler-optional.js

```diff
@@ -0,0 +1,57 @@
+function readOptionalNbt (buffer, offset, { tagType } = { tagType: 'nbt' }, rootNode) {
```

Comment:
let's remove compiler from the name since it provides both version / and let's do the same for other files